{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Datasets for the book\n",
    "\n",
    "Here we provide links to the datasets used in the book.\n",
    "\n",
    "*Note that these datasets are provided on external servers by third parties*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Next-generation Sequencing (NGS)\n",
    "\n",
    "##Working with modern sequence formats\n",
    "* SRR003265.filt.fastq.gz\n",
    "Please FTP from this URL (cut and paste)\n",
    "ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/NA18489/sequence_read/SRR003265.filt.fastq.gz\n",
    "\n",
    "##Working with modern sequence formats\n",
    "* NA18490_20_exome.bam\n",
    "Please FTP from this URL (cut and paste)\n",
    "ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/NA18489/exome_alignment/NA18489.chrom20.ILLUMINA.bwa.YRI.exome.20121211.bam\n",
    "\n",
    "* NA18490_20_exome.bam.bai\n",
    "Please FTP from this URL (cut and paste)\n",
    "ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/NA18489/exome_alignment/NA18489.chrom20.ILLUMINA.bwa.YRI.exome.20121211.bam.bai\n",
    "\n",
    "##Analyzing data in Variant Call Format (VCF)\n",
    "\n",
    "* ftp://ftp-\n",
    "trace.ncbi.nih.gov/1000genomes/ftp/release/20130502/supporting/vcf_\n",
    "with_sample_level_annotation/ALL.chr22.phase3_shapeit2_mvncall_i\n",
    "ntegrated_v5_extra_anno.20130502.genotypes.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Genomics\n",
    "\n",
    "###Working with high-quality reference genomes\n",
    "\n",
    "* [falciparum.fasta](http://plasmodb.org/common/downloads/release-9.3/Pfalciparum3D7/fasta/data/PlasmoDB-9.3_Pfalciparum3D7_Genome.fasta)\n",
    "\n",
    "###Dealing with low low-quality genome references\n",
    "\n",
    "\n",
    "* gambiae.fa.gz\n",
    "Please FTP from this URL (cut and paste)\n",
    "ftp://ftp.vectorbase.org/public_data/organism_data/agambiae/Genome/agambiae.CHROMOSOMES-PEST.AgamP3.fa.gz\n",
    "\n",
    "* [atroparvus.fa.gz](https://www.vectorbase.org/download/anopheles-atroparvus-ebroscaffoldsaatre1fagz)\n",
    "\n",
    "\n",
    "###Traversing genome annotations\n",
    "\n",
    "* [gambiae.gff3.gz](http://www.vectorbase.org/download/anopheles-gambiae-pestbasefeaturesagamp42gff3gz)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##PopGen\n",
    "\n",
    "###Managing datasets with PLINK\n",
    "\n",
    "* [hapmap.map.bz2](http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/hapmap3/plink_format/draft_2/hapmap3_r2_b36_fwd.consensus.qc.poly.map.bz2)\n",
    "* [hapmap.ped.bz2](http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/hapmap3/plink_format/draft_2/hapmap3_r2_b36_fwd.consensus.qc.poly.ped.bz2)\n",
    "* [relationships.txt](http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/hapmap3/plink_format/draft_2/relationships_w_pops_121708.txt)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##PDB\n",
    "\n",
    "###Parsing mmCIF files with Biopython\n",
    "\n",
    "* [1TUP.cif](http://www.rcsb.org/pdb/download/downloadFile.do?fileFormat=cif&compression=NO&structureId=1TUP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
